sql/core/src/main/scala/org/apache/spark/sql/execution/SparkPlan.scala

```diff
@@ -106,7 +106,27 @@ abstract class SparkPlan extends QueryPlan[SparkPlan] with Logging with Serializ
   final def execute(): RDD[InternalRow] = {
     RDDOperationScope.withScope(sparkContext, nodeName, false, true) {
       prepare()
-      doExecute()
+
+      if (sqlContext.conf.wholeStageEnabled && supportCodeGen
```

Comment:
If `WholeStageCodeGen` hold a list/tree of operators, we still want to see the details of the tree.

Right now, we can have a special flag for the operator that support it.